bottles/frontend/views/new.py

```diff
@@ -176,80 +162,119 @@ def set_path(_dialog, response):
+        """ Updates widgets based on whether it succeeded or failed. """
+
+        def send_notification(notification_id: str) -> None:
+            """ Sends notification if out of focus. """
+            if not self.is_active():
+                self.app.send_notification(notification_id, notification)
+
+        self.status_statuses.set_description(None)
+        self.is_closable = True
+        notification = Gio.Notification()
+
+        # Show error if bottle unsuccessfully builds
         if not result or not result.status or error:
-            self.update_output(_("There was an error creating the bottle."))
+            title = _("Unable to Create Bottle")
             self.btn_cancel.set_visible(False)
             self.btn_close.set_visible(True)
-            self.headerbar.remove_css_class("flat")
+            notification.set_title(title)
+            notification.set_body(_("Bottle failed to create with one or more errors."))
+            self.status_statuses.set_title(title)
+            self.btn_close.get_style_context().add_class("destructive-action")
+            send_notification("bottle-failed")
             return
 
-        self.new_bottle_config = result.data.get("config")
-        self.created.set_description(
-            _("\"{0}\" was created successfully.").format(
+        # Show success
+        title = _("Bottle Created")
+        description = _("\"{0}\" was created successfully.").format(
                 self.entry_name.get_text()
             )
-        )
-        self.btn_cancel.set_visible(False)
-        self.stack_create.set_visible_child_name("page_created")
 
-        '''
-        Ask the manager to check for new bottles, then update the
-        user bottles list. 
-        '''
+        notification = Gio.Notification()
+        notification.set_title(title)
+        notification.set_body(description)
+
+        self.new_bottle_config = result.data.get("config")
+        self.scrolled_output.set_visible(False)
+        self.btn_close.set_visible(True)
+        self.btn_close.get_style_context().add_class("suggested-action")
+        self.status_statuses.set_icon_name("selection-mode-symbolic")
+        self.status_statuses.set_title(title)
+        self.status_statuses.set_description(description)
+        send_notification("bottle-created")
+
+        # Ask the manager to check for new bottles,
+        # then update the user bottles' list.
         self.manager.check_bottles()
         self.window.page_list.update_bottles(show=result.data.get("config").get("Path"))
 
-    def __close_window(self, *_args):
-        """
-        This function check if an executable was passed to Bottles as
-        a command line argument. If so, it will be launched in the new
-        bottles and will close the bottle creation dialog. If there is
-        no arguments, it will simply close the dialog.
-        """
-        self.destroy()
+    def __radio_get_active(self) -> str:
```

Comment:
This should be rewritten IMO. GTK docs suggests using GAction, but I'm not sure how is it supposed to work either

Reply:
Should we skip this and deal with it later? GAction seems like a really complicated process...